### ISSUE TITLE
Nested cross validation for hyperparameter tuning

### ISSUE THREAD

**Issue by Francesco-Carlucci** [inventec-ai-center/bp-benchmark#11]
Good afternoon,
The related paper states in the hyperparameter tuning section that nested cross validation is used:

![Image](https://github.com/user-attachments/assets/692a8b5f-e438-4cbd-b909-313e4da2671f)

but running the code it seems to me that tune.py script calls solver.evaluate() that uses a single cross validation loop, as generated by get_nested_fold_idx function, here's the output of generator returned by the function:

![Image](https://github.com/user-attachments/assets/adb29135-1402-4748-9593-af1eed7925e6)

So, I'd like to ask you if every Optuna trial use a single (train, validation, test) cross validation split and if the optimized metric is the one relative to the validation set or the test set (paper says validation, but tune.py script returns metrics['test/sbp_mae']).  Is that code not updated?
Wouldn't we have a data leakage if we use test metrics, as Optuna would use the test set to set model's hyperparameters and DL architectures?

Thanks for your consideration and congratulations for the great work you've done, it has made model comparison a lot easier!
